x-pack/legacy/plugins/maps/public/layers/sources/es_search_source/es_search_source.js

```diff
@@ -219,33 +219,43 @@ export class ESSearchSource extends AbstractESSource {
     ];
   }
 
-  async _excludeDateFields(fieldNames) {
-    const dateFieldNames = (await this.getDateFields()).map(field => field.getName());
-    return fieldNames.filter(field => {
-      return !dateFieldNames.includes(field);
+  _getDocValueAndSourceFields(indexPattern, fieldNames) {
+    const docValueFields = [];
+    const sourceOnlyFields = [];
+    const scriptedFields = [];
```

Comment:
scriptedFields are never used by callers. `_getTopHits` should probably just grab it.